File: DrawDemoSwift/Old ObjC/GCSDrawDemoPrefsController.m
#import "GCSDrawDemoPrefsController.h"
#import "GCSDrawDemoDocument.h"
#import <DKDrawKit/DKObjectDrawingLayer.h>
#import <DKDrawKit/DKBSPObjectStorage.h>

@implementation GCSDrawDemoPrefsController

- (IBAction)qualityThrottlingAction:(id)sender
{
	[GCSDrawDemoDocument setDefaultQualityModulation:[sender intValue]];
}

- (IBAction)undoableSelectionAction:(id)sender
{
	[DKObjectDrawingLayer setDefaultSelectionChangesAreUndoable:[sender intValue]];
}

- (IBAction)setStorageTypeAction:(id)sender
{
	if ([sender intValue] == 0)
		[DKObjectOwnerLayer setStorageClass:[DKLinearObjectStorage class]];
	else
		[DKObjectOwnerLayer setStorageClass:[DKBSPObjectStorage class]];

	[[NSUserDefaults standardUserDefaults] setObject:NSStringFromClass([DKObjectOwnerLayer storageClass]) forKey:@"DKObjectStorageClass"];
}

- (void)awakeFromNib
{
	mQualityThrottlingCheckbox.intValue = [GCSDrawDemoDocument defaultQualityModulation];
	mUndoSelectionsCheckbox.intValue = [DKObjectDrawingLayer defaultSelectionChangesAreUndoable];
	mStorageTypeCheckbox.intValue = 0;
}

@end

File: DrawDemoSwift/GCSDrawDemoPrefsController.swift
//
//  GCSDrawDemoPrefsController.swift
//  DrawDemoSwift
//
//  Created by C.W. Betts on 12/14/17.
//  Copyright © 2017 DrawKit. All rights reserved.
//

import Cocoa
import DKDrawKit.DKObjectDrawingLayer
import DKDrawKit.DKBSPObjectStorage

final class GCSDrawDemoPrefsController: NSWindowController {
	@IBOutlet weak var qualityThrottlingCheckbox: NSButton!
	@IBOutlet weak var undoSelectionsCheckbox: NSButton!
	@IBOutlet weak var storageTypeCheckbox: NSButton!

    override func windowDidLoad() {
        super.windowDidLoad()
		
		qualityThrottlingCheckbox.state = DrawDemoDocument.defaultQualityModulation ? .on : .off
		undoSelectionsCheckbox.state = DKObjectDrawingLayer.defaultSelectionChangesAreUndoable ? .on : .off
		storageTypeCheckbox.state = .off
	}

	@IBAction func qualityThrottlingAction(_ sender: NSButton?) {
		if sender?.state == .on {
			DrawDemoDocument.defaultQualityModulation = true
		} else {
			DrawDemoDocument.defaultQualityModulation = false
		}
	}
	
	@IBAction func undoableSelectionAction(_ sender: NSButton?) {
		if sender?.state == .on {
			DKObjectDrawingLayer.defaultSelectionChangesAreUndoable = true
		} else {
			DKObjectDrawingLayer.defaultSelectionChangesAreUndoable = false
		}
	}

	@IBAction func setStorageTypeAction(_ sender: NSButton?) {
		if sender?.state == .off {
			DKObjectOwnerLayer.storageClass = DKLinearObjectStorage.self
		} else {
			DKObjectOwnerLayer.storageClass = DKBSPObjectStorage.self
		}
		
		UserDefaults.standard.set(NSStringFromClass(DKObjectOwnerLayer.storageClass), forKey: "DKObjectStorageClass")
	}
}
